src/main/java/org/jenkinsci/plugins/ewm/restrictions/AbstractUserRestriction.java

```diff
@@ -0,0 +1,33 @@
+package org.jenkinsci.plugins.ewm.restrictions;
+
+import hudson.model.Cause;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import org.jenkinsci.plugins.ewm.DiskPoolRestriction;
+
+import javax.annotation.Nonnull;
+
+/**
+ * @author Alexandru Somai
+ */
+public abstract class AbstractUserRestriction extends DiskPoolRestriction {
+
+    @Override
+    public boolean isAllowed(@Nonnull Run<?, ?> run, @Nonnull TaskListener listener) {
+        Cause.UserIdCause userIdCause = run.getCause(Cause.UserIdCause.class);
```

Comment:
There is a corner-case, which may impact compatibility with other plugins. In old Hudson instances/plugins there was another cause - UserCause. It's deprecated for a long time, so you can just ignore it